pkg/karmadactl/cmdinit/karmada/deploy.go

```diff
@@ -46,12 +47,13 @@ func InitKarmadaResources(dir, caBase64, systemNamespace string) error {
 		return err
 	}
 
-	// create namespace
-	if _, err := clientSet.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{
+	ns := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: systemNamespace,
 		},
-	}, metav1.CreateOptions{}); err != nil {
+	}
+	// create namespace
+	if _, err := clientSet.CoreV1().Namespaces().Create(context.TODO(), ns, metav1.CreateOptions{}); err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
I'm sure this patch could fix this issue. 
This patch blinds the error, I hope we can figure out the root cause before we make a fix.

Reply:
I test locally. It works. The root cause for it is it has been saved in etcd.

Reply:
> The root cause for it is it has been saved in etcd.

Do you mean the `deinit` forgot to remove the `karmada-system` namespace?  Is it still in the etcd?

Reply:
It is right.

Reply:
However, `deinit` not only doesn't remove ns `karmada-system`, there are some crds installed also not have been removed, which I also checks for.

Reply:
It is in `karmada-etcd`. There is exactly a problem that it can't be executed secondly after the first execution fail. I suggest to use `CreateOrUpdate()` function to create these resources, which is clearer. Like

https://github.com/karmada-io/karmada/blob/1af078cd4a97b01924e158a93246eb82a5f3af32/pkg/karmadactl/cmdinit/utils/rbac.go#L91-L105

Reply:
It seems better, but the handler of `etcd data` is still a problem. Which we use the data before or clean up and generadate data for it or someway others.

Reply:
Yeah, I get it. The root cause is clear now.

I agree with @lonelyCZ, which seems cleaner. But, can we install karmada on a `legacy etcd`?  What if there are also other resources in it? like `PP`/`Deployments`..., they will be propagated to registered clusters that might be surprised by users.

I wonder if we can clean up the default host-data-path(`/var/lib/karmada-etcd`) in `karmada init`?

Reply:
> Which we use the data before or clean up and generadate data for it or someway others.

It is enough to manage the resources of `init` phase. Ohter resources should be decided by users.

Reply:
> I wonder if we can clean up the default host-data-path(/var/lib/karmada-etcd) in karmada init?

Actually, for `PUSH` mode cluster, it can continue to work, which is benefit to upgrade karmada components.

Reply:
To install by scripts can demostrate. It can be reinstall by scripts without removing `/var/lib/karmada-etcd`, I tested it before.

Reply:
sorry, I'm still confused. 


From another perspective, does the user expected to `re-use` the legacy data in ETCD? I guess not.

By the way, do you know what's the behavior in `kubeadm`? What can we learn from it?

Reply:
> Perhaps, we also make deinit to delete local ETCD by default

Do you know why we didn't remove the etcd data before? In addition, we added a prompt at #2466 :)

> And for init, add a pre-flight phase.

Seems we don't have the `pre-fflight` thing yet, do you think we should make a quick fix? just like:
https://github.com/karmada-io/karmada/blob/1af078cd4a97b01924e158a93246eb82a5f3af32/pkg/karmadactl/cmdinit/kubernetes/deploy.go#L173-L178

Yeah, I agree, the pre-flight would be great.

Reply:
I am willing to do my best to make it better and I would appreciate it if you could provide some guidance.